### ISSUE TITLE
Is National Rail integration not working? 

### ISSUE THREAD

**Issue by mohakkataria** [morosanmihail/HA-LondonTfL#31]
I see the sensor as "None" with occasional value as Cannot reach TFL

<img width="909" alt="Screenshot 2024-07-24 at 2 36 38 PM" src="https://github.com/user-attachments/assets/b2046b50-489c-4acf-b870-e70b6dfc01fb">

These are the logs:

```
Logger: custom_components.london_tfl.sensor
Source: custom_components/london_tfl/sensor.py:184
integration: London TfL ([documentation](https://github.com/morosanmihail/HA-LondonTfL), [issues](https://github.com/morosanmihail/HA-LondonTfL/issues))
First occurred: 11:25:21 AM (8 occurrences)
Last logged: 1:51:50 PM

There was no reply from TfL servers.
```

**Comment by morosanmihail** [comment]
Yeah, not sure if thameslink would count as part of TfL. I have found that many stations show up in the api, but do not have any realtime data. But it should not be None, that hints at a connection problem to the TfL servers. They are quite finnicky.

**Comment by mohakkataria** [comment]
I think the data is available but at different endpoint for Elizabeth, national-rail and overground https://api.tfl.gov.uk/swagger/ui/index.html?url=/swagger/docs/v1#!/StopPoint/StopPoint_ArrivalDepartures

e.g. the station I was looking at : https://api.tfl.gov.uk/StopPoint/910GLBGHJN/ArrivalDepartures?lineIds=thameslink but if you try to find it in the arrivals URL: https://api.tfl.gov.uk/Line/thameslink/Arrivals/910GLBGHJN?direction=all, it returns empty

**Comment by morosanmihail** [comment]
Ah, I see! So thameslink operates more like a Bus than an underground train. I might have some time later this week to explore, but I don't think it's too difficult to fix.

**Comment by mohakkataria** [comment]
Awesome. Thanks so much :) I wish I had time otherwise I would raised an MR already. But thanks thanks so much. :)

**Comment by morosanmihail** [comment]
Sadly, said 'some amount of time' has disappeared, but was still able to take a look this morning. 

So I'll document what I found out for future me, or any future someone else. 

Buses use `/StopPoint/<>/Arrivals`. 
Thameslink (and maybe other lines) need `/StopPoint/<>/ArrivalDepartures?lineIds=<>`. 
Which is obviously different from `/Line/<>/Arrivals/<>`.

So it's not as simple as turning "is_bus" into checking if line is part of a new const that would start with `["bus", "thameslink"]`. and needs a bit more attention.

**Comment by benmarriner20** [comment]
I would be so grateful if someone with much more technical nouse than me could take up the challenge of making the National Rail arrivals work. I'm on a Southeastern line and getting this into Home Assistant would be a game changer for my morning routine!

**Comment by christopherpearson** [comment]
I've got Underground / Overground / Elizabeth line all of it working perfectly .. makes such a difference for the morning routine. THANKYOU 
However the South eastern trains are not picking up any times, see attached screenshot

<img width="704" alt="Screenshot 2024-12-19 at 10 48 26" src="https://github.com/user-attachments/assets/2e20f67f-339d-43cf-adb7-78b0073f690a" />

**Comment by morosanmihail** [comment]
Yeah, non-TfL trains do not show up. The API still shows the stations, but does not have live arrivals.
